inflect.py

```diff
@@ -2199,22 +2199,26 @@ def inflect(self, text):
     # ## PLURAL SUBROUTINES
 
     def postprocess(self, orig, inflected):
-        if "|" in inflected:
-            inflected = inflected.split("|")[self.classical_dict["all"]]
-        result = inflected.split(" ")
-        # Try to fix word wise capitalization
-        for index, word in enumerate(orig.split(" ")):
-            if word == "I":
-                # Is this the only word for exceptions like this
-                # Where the original is fully capitalized
-                # without 'meaning' capitalization?
-                # Also this fails to handle a capitalizaion in context
-                continue
-            if word.capitalize() == word:
-                result[index] = result[index].capitalize()
-            if word == word.upper():
-                result[index] = result[index].upper()
-        return " ".join(result)
+        try:
+            if "|" in inflected:
+                inflected = inflected.split("|")[self.classical_dict["all"]]
+            result = inflected.split(" ")
+            # Try to fix word wise capitalization
+            for index, word in enumerate(orig.split(" ")):
+                if word == "I":
+                    # Is this the only word for exceptions like this
+                    # Where the original is fully capitalized
+                    # without 'meaning' capitalization?
+                    # Also this fails to handle a capitalizaion in context
+                    continue
+                if word.capitalize() == word:
+                    result[index] = result[index].capitalize()
+                if word == word.upper():
+                    result[index] = result[index].upper()
+            return " ".join(result)
+        # Might occur on weird inputs, e.g. "lens with a lens ()."
+        except IndexError:
```

Comment:
This exception trap feels perhaps too broad. I'd rather the trap happen narrowly around a line or two. Or maybe there should be a test for orig vs. inflected to detect when `split(" ")` will not produce compatible arrays. Or even better would be to determine why postprocess would be called with incompatible parameters and disallow that.